gmail/src/models/email.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Represent the current email open in the Gmail application.
+ */
+export class Email {
+    messageId: string;
+    subject: string;
+    contactFullEmail: string;
+    contactName: string;
+    contactEmail: string;
+    body: string;
+
+    constructor(messageId: string = null) {
+        if (messageId) {
+            const userEmail = Session.getEffectiveUser().getEmail().toLowerCase();
+
+            this.messageId = messageId;
+            const message = GmailApp.getMessageById(this.messageId);
+            this.subject = message.getSubject();
+
+            const fromHeaders = message.getFrom();
+            const sent = fromHeaders.toLowerCase().indexOf(userEmail) >= 0;
+
+            this.contactFullEmail = sent ? message.getTo() : message.getFrom();
+            [this.contactName, this.contactEmail] = this._parseEmailAdress(this.contactFullEmail);
+
+            this.body = message.getBody();
+        }
+    }
+
+    /**
+     * Parse a full FROM header and return the name part and the email part.
+     *
+     * E.G.
+     *     "STD" <std@odoo.com> => ["STD", "std@odoo.com"]
+     *     std@odoo.com         => ["std@odoo.com", "std@odoo.com"]
+     *
+     */
+    _parseEmailAdress(fullEmail: string): [string, string] {
```

Comment:
It might be "technically" possible, but it will require to add a button in the debug view ("Run tests") which will execute a "test function". But it will be useful only to test that method, so is a bit overkill no ?

Reply:
If there is no automation aspect to it then it's a bit useless yes.
Thanks for checking anyway.